controllers/api/domain/domain.go

```diff
@@ -114,7 +118,10 @@ func (g *Controller) sync(key string, svc *v1.Service) error {
 
 	if len(ips) == 0 {
 		ep, err := g.endpointsLister.Get(svc.Namespace, svc.Name)
-		if err != nil {
+		if err != nil && errors.IsNotFound(err) {
+			g.serviceController.Enqueue(svc.Namespace, svc.Name)
```

Comment:
are you totally sure you won't get in an infinite loop?  Is it possible that we might just want to mask this error and just return nil on NotFound